lib/signaling/participant.js

```diff
@@ -198,14 +208,30 @@ class ParticipantSignaling extends StateMachine {
    * @returns {boolean}
    */
   connect(sid, identity) {
-    if (this.state === 'connecting') {
-      this._sid = sid;
-      this._identity = identity;
+    if (this.state === 'connecting' || this.state === 'reconnecting') {
+      if (!this._sid) {
+        this._sid = sid;
+      }
+      if (!this._identity) {
+        this._identity = identity;
+      }
       this.preempt('connected');
       return true;
     }
     return false;
```

Comment:
is this case possible - that we get connected when not 'connecting' or 'reconnecting' ? Maybe it's worth `log.error("unexpected connect message")`

Reply:
@makarandp0 No, it is not possible, unless there is a bug in Room Service. So I don't think we need to log the error.

Reply:
yeah it makes sense that it's not possible, I would suggest we remove the if statement that guards for it
`if (this.state === 'connecting' || this.state === 'reconnecting') {`  
But if we are keeping it - which is okay too - an assertion or log.error helps make it clear to reader that this is not expected.